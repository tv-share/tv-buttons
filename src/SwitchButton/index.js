import "./switchButton.*";
import SwitchButton from "./SwitchButton";

export default SwitchButton;