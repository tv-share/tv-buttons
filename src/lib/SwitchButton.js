import React from "react";
import PropTypes from "prop-types";

/**
* Renders the Switch Component
* @param  {String}    	className is a string that will be attached to the select input's className (optional)
* @param  {Boolean}     enabled specifies if the button is currently on enabled position
* @param  {Function}    onClick defines a function to be called whenever the button is pressed
* @param  {Function}    onLabel defines a label for the ON position (right)
* @param  {Function}    offLabel defines a label for the OFF position (left)
* @return {JSX}         Switch Component
*/
const SwitchButton = ({ className, enabled, onClick, onLabel = "", offLabel = "" }) => {
    return (
        <div className="tv-switch-btn">
            <label>
                {offLabel}
                <input checked={enabled} onChange={onClick} type="checkbox" />
                <span className="tv-lever"></span>
                {onLabel}
            </label>
        </div>
    );
};

SwitchButton.propTypes = {
    enabled: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    onLabel: PropTypes.string,
    offLabel: PropTypes.string
};

export default SwitchButton;