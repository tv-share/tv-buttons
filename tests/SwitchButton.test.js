import React from 'react';
import renderer from 'react-test-renderer';

import SwitchButton from '../src/lib/SwitchButton';

const fn = () => { };

it('renders correctely on ON position', () => {
	const component = renderer.create(
		<SwitchButton enabled={true} onClick={fn}/>
	);

	expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctely on OFF position', () => {
	const component = renderer.create(
		<SwitchButton enabled={false} onClick={fn}/>
	);

	expect(component.toJSON()).toMatchSnapshot();
});