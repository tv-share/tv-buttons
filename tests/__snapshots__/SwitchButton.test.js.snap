// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctely on OFF position 1`] = `
<div
  className="tv-switch-btn"
>
  <label>
    
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="tv-lever"
    />
    
  </label>
</div>
`;

exports[`renders correctely on ON position 1`] = `
<div
  className="tv-switch-btn"
>
  <label>
    
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="tv-lever"
    />
    
  </label>
</div>
`;
